aws-lc-rs/src/fips.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+/// Retrieve the FIPS module service status.
+#[must_use]
+pub fn get_fips_service_status() -> FipsServiceStatus<()> {
+    #[cfg(feature = "fips")]
+    {
+        if indicator::is_approved() {
+            FipsServiceStatus::ApprovedMode(())
+        } else {
+            FipsServiceStatus::NonApprovedMode(())
+        }
+    }
+    #[cfg(not(feature = "fips"))]
+    FipsServiceStatus::NotEnabled(())
+}
+
+/// Clears the FIPS module service status. Useful if a thread has executed an unapproved
+/// algorithm, but you desire to reset the module status without destroying the thread.
+#[cfg(feature = "fips")]
+pub fn reset_fips_service_status() {
+    indicator::reset();
```

Comment:
There could be a "race" on this.

Not a "race" in the typical definition. But an application that is build with two different libraries that both use aws-lc-rs. One resets, while the other relies on an error being thrown later.

Reply:
Another thing to consider is nested scopes. If you wrap a `check_fips_service_status` inside of another `check_fips_service_status`, do you end up with a weird state?